From ade9712641309bf14c1b6e50290ff331abc0f313 Mon Sep 17 00:00:00 2001
From: Steev Klimaszewski <steev@kali.org>
Date: Tue, 20 Sep 2022 12:59:49 -0500
Subject: [PATCH 26/29] drm/meson: Hold fewer lines after vsync

On newer SoCs we need to hold more lines for AFBC support. See commit
24e0d40 ("drm/meson: hold 32 lines after vsync to give time for
AFBC start") for an in-depth explanation.

However, for some reason, this causes other side effects like screen
tearing every so often.  Endless' 3.10 kernel has this lowered to 12,
however, locally I needed to go back to the original 4.

Signed-off-by: Steev Klimaszewski <steev@kali.org>
---
 drivers/gpu/drm/meson/meson_viu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/meson/meson_viu.c b/drivers/gpu/drm/meson/meson_viu.c
index d4b907889a21..0d6601db1623 100644
--- a/drivers/gpu/drm/meson/meson_viu.c
+++ b/drivers/gpu/drm/meson/meson_viu.c
@@ -436,7 +436,7 @@ void meson_viu_init(struct meson_drm *priv)
 
 	/* Initialize OSD1 fifo control register */
 	reg = VIU_OSD_DDR_PRIORITY_URGENT |
-		VIU_OSD_HOLD_FIFO_LINES(31) |
+		VIU_OSD_HOLD_FIFO_LINES(4) |
 		VIU_OSD_FIFO_DEPTH_VAL(32) | /* fifo_depth_val: 32*8=256 */
 		VIU_OSD_WORDS_PER_BURST(4) | /* 4 words in 1 burst */
 		VIU_OSD_FIFO_LIMITS(2);      /* fifo_lim: 2*16=32 */
-- 
2.38.1

